esp-hal/src/lib.rs

```diff
@@ -398,6 +398,18 @@ impl Cpu {
             }
         }
     }
+
+    /// Returns an iterator over all cores.
+    #[inline(always)]
+    pub fn all() -> impl Iterator<Item = Self> {
```

Comment:
Don't feel strongly about this so feel free to ignore, but maybe `iter()` is a bit more idiomatic?

Reply:
My thinking was that we already have `other` which is also an iterator, it's just `all - current`. Although I guess I don't have to make this (or Cpu::other as I don't think we've released it yet) immediately public.